app/src/main/java/com/limelight/Game.java

```diff
@@ -575,8 +575,13 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
         }
 
         boolean handled = false;
-        if (event.getDevice() == null ||
-                event.getDevice().getKeyboardType() != InputDevice.KEYBOARD_TYPE_ALPHABETIC) {
+
+        boolean detectedGamepad = event.getDevice() == null ? false :
+                ((event.getDevice().getSources() & InputDevice.SOURCE_JOYSTICK) != 0 ||
+                        (event.getDevice().getSources() & InputDevice.SOURCE_GAMEPAD) != 0);
+        if (detectedGamepad || (event.getDevice() == null ||
+                event.getDevice().getKeyboardType() != InputDevice.KEYBOARD_TYPE_ALPHABETIC
+        )) {
```

Comment:
I don't see how this should change the existing behavior. Is the ipega marked as `InputDevice.KEYBOARD_TYPE_ALPHABETIC`?